src/symbols_linux.cpp

```diff
@@ -845,4 +836,48 @@ void Symbols::parseLibraries(CodeCacheArray* array, bool kernel_symbols) {
     }
 }
 
+static bool isValidHandle(const CodeCache* cc, void* handle) {
+    Dl_info dl_info;
+    struct link_map* map;
+
+    // Check shared objects are still available at original base image
+    if (handle == NULL || dlinfo(handle, RTLD_DI_LINKMAP, &map) != 0 || dladdr(map->l_ld, &dl_info) == 0) {
+        return false;
+    }
+
+    return cc->imageBase() == (const char*)dl_info.dli_fbase;
+}
+
+UnloadProtection::UnloadProtection(const CodeCache *cc) {
+    _lib_handle = NULL;
+    _valid = false;
+
+    if (OS::isMusl() || isMainExecutable(cc->imageBase(), cc->maxAddress()) || isLoader(cc->imageBase())) {
+        _valid = true;
+        return;
+    }
+
+    const char* stripped_name;
+    size_t name_len = strlen(cc->name());
+    if (name_len > 10 && strcmp(cc->name() + name_len - 10, " (deleted)") == 0) {
+        char* buf = (char*) alloca(name_len);
+        memcpy(buf, cc->name(), name_len);
+        buf[name_len - 10] = 0;
+        stripped_name = buf;
+    } else {
+        stripped_name = cc->name();
+    }
```

Comment:
I'm trying to write the same in a shorter way. WDYT about the following?
```c
    const char* stripped_name = cc->name();
    size_t name_len = strlen(stripped_name);
    if (name_len > 10 && strcmp(stripped_name + name_len - 10, " (deleted)") == 0) {
        char* buf = (char*) alloca(name_len);
        *stpncpy(buf, stripped_name, name_len - 10) = 0;
        stripped_name = buf;
    }
```

Reply:
Looks good, `buf` can also be `name_len - 9`

Reply:
9ae2c0f512356df2c4441841aaf3a28a010f088f